proposed/2023/NuGetAudit-without-nuget.org-as-packageSource.md

```diff
@@ -0,0 +1,221 @@
+When there is at least one source in `auditSources`, NuGet should no longer use `packageSources` to get a VDB.
+Omitting `<clear />` will not use `packageSources` in addition to `auditSources`.
+
+For example, consider a developer at Contoso using a company-internal source as a single package source, and using nuget.org as an audit source:
+
+```xml
+<configuration>
+  <packageSources>
+    <!-- Clear to ensure package sources not inherited from other config files -->
+    <clear />
+    <add key="contoso" value="https://internal.corp.contoso.com/packages/nuget/index.json" >
+  </packageSources>
+  <auditSources>
+    <!-- Clear to ensure audit sources not inherited from other config files -->
+    <clear />
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" >
+  </auditSources>
+</configuration>
+```
+
+Any URL specified in the `auditSources` section must implement the [NuGet Server API V3 protocol](https://learn.microsoft.com/nuget/api/overview).
+However, the server will only need to implement the [`VulnerabilityInfo` resource](https://learn.microsoft.com/en-us/nuget/api/vulnerability-info) (and service index), and can avoid implementing all the other resources that are required to be used as a package source.
+
+NuGet will use same semantics for `auditSources` as it already has with `packageSources`.
+This means that `auditSources` are accumulated from other nuget.config files, and `<clear />` can be used to remove them.
+`auditSources` will also follow `packageSources` with regards to insecure http warnings, errors, and `allowInsecureConnections` configurations.
+Sources that need credentials will use `packageSourceCredentials`, using the `key` as the lookup, just as `packageSources` does.
+This means that credentials can be re-used by using the same key for a source that is defined in both `packageSources` and `auditSources`.
+
+#### Restore summary
+
+When restore is run at normal or detailed verbosity, NuGet outputs detected nuget.config files and package sources (feeds) used.
+Audit sources should be added.
+
+```diff
+ NuGet Config files used:
+     C:\Users\zivkan\AppData\Roaming\NuGet\NuGet.Config
+
+ Feeds used:
+     https://internal.corp.contoso.com/packages/nuget/index.json
++
++Audit sources used:
++    https://api.nuget.org/v3/index.json
+```
+
+The "Audit sources used" section should ideally list sources where a VDB was successfully obtained.
+This depends on the feasibility of implementation.
+For example, if two package sources are defined, and no audit sources are defined by the *nuget.config* file, but only one of the two sources has a VDB, then the restore summary will only list the one source with a VDB under "Audit sources used".
+
+If `NuGetAudit` is disabled for all projects in the solution, the audit sources used section of the summary should be omitted.
+
+### Technical explanation
```

Comment:
Will the HTTP cache for the audit source be shared with the package source of the same URL?

Reply:
yes, as written in the functional explanation section, audit sources use all the same semantics as package sources.